plugins/headless/README.md

```diff
@@ -1,13 +1,101 @@
-# eclipse-headless-project-import-tool
-Imports an eclipse project from archive file into an existing workspace
+# Headless Gamma service
 
-HOW TO USE:
+This document details the usage of the Headless Gamma service, which allows the execution of `.ggen` files using a _webserver_ based on OpenAPI and an _exported headless Eclipse_. The service can be used via Docker. Alternatively, it is possible to manually export the headless Eclipse and start the webserver.
 
-Import this project in your eclipse IDE and extract the product using the Product Export Wizard.
-You should get a headless eclipse.
-Pass the workspace as -data and the project name which should be the same as the archive file name, the archive file should already be in the workspace.
-Example:
-eclipse -consoleLog -data C:\workspace1 project.myfirstproject
+This feature uses two projects:
 
-Tip:
-if you want to create a new workspace just pass the -data argument with the name of the workspace you want to create and do not pass any project names.
+
+ - `hu.bme.mit.gamma.headless.api` - this application serves as the headless version of Gamma. It can create Eclipse workspaces, import projects from zip files, and execute .ggen files.
+ - `hu.bme.mit.gamma.headless.server` - this webserver forwards requests to the Headless Gamma feature exported from `hu.bme.mit.gamma.headless.api`.
+
+Note that this document serves as a high level description of the workflow.  A more detailed, technical documentation regarding the manual exporting of the features and the creation of Docker images can be found in the`docs` folder of the `hu.bme.mit.gamma.headless.api` project.
+
+
+## Requirements
+
+As mentioned before, the Headless Gamma service can be used either via Docker or by the manual exporting of the Gamma features.
+
+- To use this feature via Docker, you have to have Docker installed on your computer, and download the image available here:
+https://hub.docker.com/repository/docker/ftsrggamma/headless-gamma
+
+- To manually export Gamma, first create a functioning Eclipse with Gamma by following the process detailed here: https://github.com/ftsrg/gamma/tree/master/plugins. Note that currently this feature uses the `dev` branch of Gamma.
+

```

Comment:
Delete unnecessary new lines, only one should remain.

Reply:
Deleted unnecessary new lines.